components/server/src/prometheus-metrics.ts

```diff
@@ -381,3 +381,27 @@ type AuthorizerSubjectIdMatch = "ctx-user-id-missing" | "passed-subject-id-missi
 export function reportAuthorizerSubjectId(match: AuthorizerSubjectIdMatch) {
     authorizerSubjectId.labels(match).inc();
 }
+
+export const scmTokenRefreshRequestsTotal = new prometheusClient.Counter({
+    name: "gitpod_scm_token_refresh_requests_total",
+    help: "Counter for the number of token refresh requests we issue against SCM systems",
+    labelNames: ["host", "result"],
```

Comment:
Will `host` cause 💸 on Gitpod Cloud? Cost of grafana is based num of values of labels too

Reply:
In general both metrics are rather costly. I have a todo to check how much it is in the end.

:thought_balloon: I wonder whether we should have means to enable/disable detailed reporting like this at runtime, either with a feature flag or similar.

Reply:
Let's see how it goes after deploy